src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
-import './App.css';
+import React, {useEffect} from 'react';
+import {
+  RouterProvider,
+} from "react-router-dom";
+import {useAppDispatch} from "./Hooks/hook";
+import auth from "./Firebase";
+import {userAuth, userLogOut, userToken} from "./Redux/Store/auth/authSlice";
+import axios from "axios";
+import {appRouter} from "./Routes/Router";
+
+
+function App():JSX.Element {
+  const dispatch = useAppDispatch()
+  useEffect(()=> {
+    auth.onAuthStateChanged(authUser => {
+      if(authUser){
+        // console.log(authUser);
+        authUser.getIdTokenResult().then(result=> {
+          // console.log(result);
+          const token:string = result.token
+          dispatch(userToken(token))
+          const request: Promise<void> = axios.get("http://localhost:3000/userAuth/verifyUser",{
+            headers:{
+              Authorization: `Bearer ${token}`
+            }
+          }).then(response => {
+            const data = response.data
+            dispatch(userAuth({
+              uId: data.user_id,
+              email: data.email,
+              name: data.name,
+              picture: data.picture
+            }))
+          }).catch(e => console.log(e.message))
+        }).catch(error => console.error("something wrong: " + error.message));
```

Comment:
Here some error handling instead of logging the error looks better

Reply:
Agree! I did it using Toastify notifications.